src/main/java/org/wso2/carbon/esb/connector/MQPublishTopic.java

```diff
@@ -0,0 +1,134 @@
+        try {
+
+            log.info("Message received at ibm-mq connector");
+            connectionBuilder = new MQConnectionBuilder(messageContext);
+            config = connectionBuilder.getConfig();
+
+            SOAPEnvelope soapEnvelope = messageContext.getEnvelope();
+            OMElement getElement = soapEnvelope.getBody().getFirstElement();
+            String queueMessage = "";
+
+            if (getElement != null) {
+                queueMessage = getElement.toString();
+            }
+
+            MQMessage mqMessage = buildMQMessage(config, queueMessage);
+
+            if (config.getTopicName() != null) {
+                topic = setTopic(connectionBuilder, config);
+            }
+
+            if (topic == null || mqMessage == null) {
+                log.error("Error in publishing message");
+            } else {
+                log.info("topic initialized");
+                topic.put(mqMessage);
+                log.info("Message successfully placed at " + config.getTopicName() + "topic");
+            }
+
+            connectionBuilder.closeConnection();
+
+        } catch (Exception e) {
+            log.error("Problem in publishing to message" + e);
+            throw new ConnectException(e);
+        }
+    }
+
+    /**
+     * Initialize topic
+     */
+    MQTopic setTopic(MQConnectionBuilder connectionBuilder, MQConfiguration config) {
+        MQQueueManager manager = connectionBuilder.getManager();
+        MQTopic publisher;
+        try {
+            publisher = manager.accessTopic(config.getTopicString(), config.getTopicName(),
+                    CMQC.MQTOPIC_OPEN_AS_PUBLICATION, CMQC.MQOO_OUTPUT);
+        } catch (MQException e) {
+            log.error("Error creating topic" + e);
+            return null;
+        }
+        return publisher;
+    }
+
+    /**
+     * Create MQMessage
+     */
+    MQMessage buildMQMessage(MQConfiguration config, String queueMessage) {
+
+        MQMessage mqMessage = new MQMessage();
+        mqMessage.messageId = config.getMessageID().getBytes();
+        mqMessage.correlationId = config.getCorrelationID().getBytes();
+        if (config.getPriority() != -1000) {
+            mqMessage.priority = config.getPriority();
+        }
+        int messageType = config.getMessageType();
+        mqMessage.messageType = messageType;
+
+        if (config.isPersistent()) {
+            mqMessage.persistence = MQPER_PERSISTENT;
+        } else {
+            mqMessage.persistence = MQPER_NOT_PERSISTENT;
+        }
+        if (config.getgroupID() != null) {
+            mqMessage.groupId = config.getgroupID().getBytes();
+        }
+
+        GregorianCalendar cal = new GregorianCalendar();
+        cal.setTime(new Date(System.currentTimeMillis()));
+        mqMessage.putDateTime = cal;
+
+        try {
+            mqMessage.writeString(queueMessage);
+        } catch (Exception e) {
+            log.info("Error creating mq message" + e);
```

Comment:
Returning null and handle error as generic, we will be loosing information why this got failed. It s better if we could throw this exception to higher layer and handle over there.